src/github.com/getlantern/flashlight/pubsub/pubsub_test.go

```diff
@@ -0,0 +1,32 @@
+package pubsub
+
+import (
+	"runtime"
+	"testing"
+)
+
+type str struct {
+	Val string
+}
+
+func TestSub(t *testing.T) {
+	if Sub(&str{}, func(teststr *str) {}) != nil {
+		t.Fail()
+	}
+	if Sub(&str{}, "String") == nil {
+		t.Fail()
+	}
+}
+
+func TestPublish(t *testing.T) {
+	var received string = ""
+	Sub(&str{}, func(s *str) {
+		received = s.Val
+	})
+	Pub(&str{Val: "test"})
+
+	runtime.Gosched()
+	if received != "test" {
```

Comment:
`received` is mutated in a different goroutine without synchronization.  Although apparently it works now, I don't think Go guarantees any kind of consistency here.  I'd signal success through a channel with a short timeout instead.